url2df0f7c811_przemekgalazka_legacycode_tgz-pJ8-pl_cc_core_cmd_CommandJ8/src/ccproxy/Events.java

```diff
@@ -0,0 +1,249 @@
+		p.agenci.pushSuperMessage(eventMsg);
+	}
+
+	public void loginIncorrect(AgentConnection ac) {
+		String eventMsg = new String(
+				"EVENT [Agent], Action [Incorect login], Agent ["
+						+ ac.agentNumber + "], Exten [" + ac.agentExten + "]"
+						+ ", Name [" + ac.agentName + "]" + ", ID ["
+						+ ac.idConnection + "]");
+		log.info(eventMsg);
+		p.agenci.pushSuperMessage(eventMsg);
+
+	}
+
+	public void removeQueueFailed(AgentConnection ac) {
+		String eventMsg = new String(
+				"EVENT [Agent], Action [Remove queue failed], Agent ["
+						+ ac.agentNumber + "], Exten [" + ac.agentExten + "]"
+						+ ", Name [" + ac.agentName + "]" + ", ID ["
+						+ ac.idConnection + "]");
+		log.info(eventMsg);
+		p.agenci.pushSuperMessage(eventMsg);
+
+	}
+
+	public void pause(AgentConnection ac, String pauseType, int pauseTime, int timePoor, int timeBad) {
+		String eventMsg = new String("EVENT [Agent], Action [Pause], Agent ["
+				+ ac.agentNumber + "], Exten [" + ac.agentExten + "]"
+				+ ", Name [" + ac.agentName + "]" + ", ID [" + ac.idConnection+ "]" +
+						", Type ["+ pauseType +"], Time [" + pauseTime + "]");
+        if (p.pauseTypeList.findById(PauseType.ID_AUTO).getName().equals(pauseType)){
+            eventMsg = eventMsg + ", TimePoor ["+timePoor+"], TimeBad ["+timeBad+"]";
+        }
+		log.info(eventMsg);
+        ac.paused=true;
+        ResultSet rs;
+        rs = p.dbConn.query("select agent_pause_ccproxy('" + ac.agentNumber +"','" + pauseType + "','" + ac.agentExten + "')");
+        try {
+            if (rs.next()) {
+            }
+        } catch (SQLException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+		p.agenci.pushSuperMessage(eventMsg);
+	
+	}
+	
+	public void pause(AgentConnection ac) {
+		pause(ac, "unknown", PauseType.PAUSE_FOREVER, 0, 0);
+	}
+
+	public void unpause(AgentConnection ac) {
+		String eventMsg = new String("EVENT [Agent], Action [Unpause], Agent ["
+				+ ac.agentNumber + "], Exten [" + ac.agentExten + "]"
+				+ ", Name [" + ac.agentName + "]" + ", ID [" + ac.idConnection
+				+ "]");
+		log.info(eventMsg);
+        ac.paused=false;
+        ResultSet rs;
+        rs = p.dbConn.query("select agent_pause_ccproxy('" + ac.agentNumber +"'," + "NULL" + ",'" + ac.agentExten +"')");
+        try {
+            if (rs.next()) {
+            }
+        } catch (SQLException e) {
+            // TODO Auto-generated catch block
+            e.printStackTrace();
+        }
+        p.agenci.pushSuperMessage(eventMsg);
+
+	}
+
+	public void addQueue(AgentConnection ac, String qname, String penalty) {
+		String eventMsg = new String(
+				"EVENT [Agent], Action [AddToQueue], Agent [" + ac.agentNumber
+						+ "], Queue [" + qname + "], Exten [" + ac.agentExten
+						+ "]" + ", Name [" + ac.agentName + "]" + ", Penalty ["
+						+ penalty + "]" + ", ID [" + ac.idConnection + "]");
+		log.info(eventMsg);
+		p.agenci.pushSuperMessage(eventMsg);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 117:123, 138:145.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.